### ISSUE TITLE
[request]: Make `***InputOperationOutputAlias` and `***InputOperationRetryAlias` public.

### ISSUE THREAD

**Issue by Hakuyume** [awslabs/aws-sdk-rust#336]
### Tell us about your request

Make `***InputOperationOutputAlias` and `***InputOperationRetryAlias` public.

### Tell us about the problem you're trying to solve.

When I write a helper function, I need put some trait boundaries on `Client`.
But the boundaries depend on private aliases.

```rust
async fn my_helper<C, M, R>(client: &Client<C, M, R>)
where
    C: SmithyConnector,
    M: SmithyMiddleware<C>,
    R: NewRequestPolicy,
    R::Policy: SmithyRetryPolicy<
        ListObjectsV2InputOperationOutputAlias,  // this alias is not public
        ListObjectsV2Output,
        ListObjectsV2Error,
        ListObjectsV2InputOperationRetryAlias,  // this alias is not public
    >, 
{
   ...
   client
       .list_objects_v2()
       ...
       .send()  // <- ListObjectsV2::send requires additional trait boundary for R::Policy
       .await
       .unwrap()
   ...
}
```

### Are you currently working around this issue?

Maybe I can replacing aliases with actual types. But It is lengthy.

### Additional context

Exposing trait alias (e.g. `ListObjectsV2SmithyRetryPolicy`) will be more useful.
But it requires [rfc1733](https://github.com/rust-lang/rfcs/blob/master/text/1733-trait-alias.md).
```rust
// more useful but impossible
pub trait ListObjectsV2SmithyRetryPolicy = SmithyRetryPolicy<
        ListObjectsV2InputOperationOutputAlias,
        ListObjectsV2Output,
        ListObjectsV2Error,
        ListObjectsV2InputOperationRetryAlias,
    >;


async fn my_helper<C, M, R>(client: &Client<C, M, R>)
where
    C: SmithyConnector,
    M: SmithyMiddleware<C>,
    R: NewRequestPolicy,
    R::Policy: ListObjectsV2SmithyRetryPolicy,  // better readability
{
    ... 
```

**Comment by Hakuyume** [comment]
I found actual types are not complex.

**Comment by Velfi** [comment]
I'm glad you found a solution. I think the bigger issue at play is that generics can make it hard for users to interact with `Client`. Hopefully the team can find a way of making that interaction less onerous.

**Comment by Velfi** [comment]
Hey @Hakuyume, I was wondering if you could explain more about what you were trying to accomplish here so that we could possibly create an example from it to help future people with the same need.

**Comment by Hakuyume** [comment]
Actually, I tried to write a helper library that handles S3's multipart-upload.
It wraps `CreateMultipartUpload`, `UploadPart`, and `CompleteMultipartUpload`.

Here is the signature of the entrypoint function. To make my wrapper general, I re-exported `C`, `M`, and `R` as generic parameters.
https://github.com/Hakuyume/s3-mpu-rs/blob/2ccb4d49cd8c142327c188ef10d6c0cb2ee6cd79/src/lib.rs#L68-L97

I have to put some trait boundaries on `R` to call `CreateMultipartUpload.send` and other aws-sdk-s3's methods.
For example, `CreateMultipartUpload` requires `R::Policy: SmithyRetryPolicy<CreateMultipartUploadInputOperationOutputAlias, CreateMultipartUploadOutput, CreateMultipartUploadError, CreateMultipartUploadInputOperationRetryAlias>`
https://docs.rs/aws-sdk-s3/latest/aws_sdk_s3/client/fluent_builders/struct.CreateMultipartUpload.html#method.send

I tried to just copy and paste this trait boundary from the docs to my code and I found `CreateMultipartUploadInputOperationOutputAlias` and `CreateMultipartUploadInputOperationRetryAlias` are private.
That's why I made this issue.

After reading the code of aws-sdk-s3, I found `CreateMultipartUploadInputOperationOutputAlias` is an alias of `CreateMultipartUpload`. I rewrote my trait boundary with this actual type, my code worked.
